contrib/inventory/ec2.py

```diff
@@ -1026,28 +952,28 @@ def add_instance(self, instance, region):
 
         # Inventory: Group by AWS account ID
         if self.group_by_aws_account:
-            self.push(self.inventory, self.aws_account_id, hostname)
+            self.push(self.inventory, self.aws_account_id, dest)
             if self.nested_groups:
                 self.push_group(self.inventory, 'accounts', self.aws_account_id)
 
         # Inventory: Group by tag keys
         if self.group_by_tag_keys:
-            for k, v in instance.tags.items():
-                if self.expand_csv_tags and v and ',' in v:
-                    values = map(lambda x: x.strip(), v.split(','))
+            for tags in instance.tags:
+                if self.expand_csv_tags and tags['Value'] and ',' in tags['Value']:
+                    values = map(lambda x: x.string(), tags['Value'].split(''))
```

Comment:
I think this should be:

```
values = map(lambda x: x.strip(), tags['Value'].split(','))
```

Basically the same as it was before this change but with `tags['Value']` instead.